iquip/apps/builder.py

```diff
@@ -1,32 +1,212 @@
+        return self.checkBox.isChecked()
+
+
+class _EnumerationEntry(_BaseEntry):
+    """Entry class for an enumeration value.
+    
+    Attributes:
+        argInfo: Each key and its value are:
+            choices: The pre-defined candidates.
+            default: The default value. If it does not exist, it is set to the first candidate.
+        comboBox: The combobox showing the enumeration value.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        argInfo: Dict[str, Any],
+        parent: Optional[QWidget] = None
+    ):
+        """Extended."""
+        super().__init__(name, argInfo, parent=parent)
+        # widgets
+        self.comboBox = QComboBox(self)
+        for choice in self.argInfo["choices"]:
+            self.comboBox.addItem(choice)
+        if self.argInfo["choices"]:
+            self.comboBox.setCurrentText(
+                self.argInfo["choices"][0] if self.argInfo["default"] is None
+                else self.argInfo["default"]
+            )
+        # layout
+        self.layout.addWidget(self.comboBox)
+
+    def value(self) -> str:
+        """Overridden.
+        
+        Returns the value of the comboBox.
+        """
+        return self.comboBox.currentText()
+
+
+class _NumberEntry(_BaseEntry):
+    """Entry class for a number value.
+    
+    Attributes:
+        argInfo: Each key and its value are:
+            default: The default value. If None, it is set to the min value.
+            unit: The unit of the value.
+            scale: The scale factor that is multiplied to the number value.
+            step: The step between values changed by the up and down button.
+            min: The minimum value. (default=0.0)
+            max: The maximum value. (default=99.99)
+            ndecimals: The number of displayed decimals.
+            type: The type of the value. If "int", value() returns an integer value.
+        spinBox: The spinbox showing the number value.
+    
+    TODO(BECATRUE): The operations of unit and scale will be concretized in Basic Runner project.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        argInfo: Dict[str, Any],
+        parent: Optional[QWidget] = None
+    ):
+        """Extended."""
+        super().__init__(name, argInfo, parent=parent)
+        # widgets
+        self.spinBox = QDoubleSpinBox(self)
+        self.spinBox.setSuffix(self.argInfo["unit"])
+        self.spinBox.setSingleStep(self.argInfo["step"] / self.argInfo["scale"])
+        if self.argInfo["min"] is not None:
+            self.spinBox.setMinimum(self.argInfo["min"] / self.argInfo["scale"])
+        if self.argInfo["max"] is not None:
+            self.spinBox.setMaximum(self.argInfo["max"] / self.argInfo["scale"])
+        self.spinBox.setDecimals(self.argInfo["ndecimals"])
+        self.spinBox.setValue((self.argInfo["min"] if self.argInfo["default"] is None \
+                               else self.argInfo["default"]) / self.argInfo["scale"])
```

Comment:
1. Is the backslash necessary?
2. The expression is too convoluted. You can extract the `if-else` expression as a variable.